mad/telegram.go

```diff
@@ -0,0 +1,97 @@
+package mad
+
+import (
+	"context"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+)
+
+type TelegramRepository struct {
+	Db *mongo.Database
+}
+
+var indexesOpts = []mongo.IndexModel{
+	{Keys: bson.M{"telegramId": 1}, Options: options.Index().SetUnique(true)},
+	{Keys: bson.M{"userId": 1}, Options: options.Index().SetUnique(true)},
+}
+
+// Get the Telegram collection and create indexes if they have not yet been
+func CreateCollection(ctx context.Context, db *mongo.Database) TelegramRepository {
+	col := db.Collection("telegram")
+
+	opts := options.ListIndexes()
+	cursor, err := col.Indexes().List(ctx, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var indexes []bson.M
+	if err = cursor.All(ctx, &indexes); err != nil {
+		log.Fatal(err)
+	}
+
+	// Assume that 1 index would be _id
+	if len(indexes) <= 1 {
```

Comment:
Assuming we have 2 indices in place, after we add third this code would not pick up the change.
What if we compare existing indices with indices in `indexesOpts` and call [CreateMany](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo?tab=doc#IndexView.CreateMany) on whatever is missing